lib/src/components/switch/Switch.tsx

```diff
@@ -26,6 +27,7 @@ const StyledSwitch = styled(RadixSwitch.Root, {
   '&[data-state="checked"]:hover': {
     backgroundColor: '$primaryMid'
   },
+  '&[disabled]': { ...disabledStyle() },
```

Comment:
```suggestion
  '&[disabled]': disabledStyle,
```

There's no need to spread the result here, we can just reference it directly. (treating it as an object as per my other comment)